keys.cabal

```diff
@@ -65,7 +105,28 @@ library
     build-depends: ghc-prim == 0.2.*
 
   if !impl(ghc >= 8.0)
-    build-depends: base-orphans >= 0.5.4 && < 0.10
+    build-depends: base-orphans >= 0.5.4 && < 0.10, semigroups >= 0.8.3.1 && < 1
+
+  if impl(ghc >= 9.6)
+    build-depends: transformers >= 0.6.1.0
+
+  if !impl(ghc >= 9.6) && impl(ghc > 7.10)
```

Comment:
Minor nitpick: can you change `impl(ghc > 7.10)` to `impl(ghc >= 8.0)`? I find it easier to read the latter, and it's more consistent with the existing version bound conventions.